model/gcp/registry.yaml

```diff
@@ -39,6 +39,75 @@ groups:
           [`CLOUD_RUN_TASK_INDEX`](https://cloud.google.com/run/docs/container-contract#jobs-env-vars)
           environment variable.
         examples: [0, 1]
+  - id: registry.gcp.apphub
+    type: attribute_group
+    display_name: GCP - AppHub Attributes
+    brief: >
+      This document defines attributes AppHub will apply to resources in GCP.
+      See [AppHub overview](https://cloud.google.com/app-hub/docs/overview).
+    attributes:
+      - id: gcp.apphub.application.container
+        type: string
+        stability: development
+        brief: >
+          The container within GCP where the AppHub application is defined.
+        examples: ['projects/my-container-project']
+      - id: gcp.apphub.application.location
+        type: string
+        stability: development
+        brief: >
+          The GCP zone or region where the application is defined.
+        examples: ['us-central1']
+      - id: gcp.apphub.application.id
+        type: string
+        stability: development
+        brief: >
+          The name of the application as configured in AppHub.
+        examples: ['my-application']
+      - id: gcp.apphub.service.id
+        type: string
+        stability: development
+        brief: >
+          The name of the service as configured in AppHub.
+        examples: ['my-service']
+      - id: gcp.apphub.service.environment_type
+        type: string
+        stability: development
+        brief: >
+          Environment of an application, service, or workload is the stage of a software lifecycle, for example, production, staging, or development.
+        note: >
+          [See AppHub properties and attributes](https://cloud.google.com/app-hub/docs/overview#properties-attributes)
+        examples: ['DEVELOPMENT', 'TEST', 'STAGING', 'PRODUCTION']
+      - id: gcp.apphub.service.criticality_type
+        type: string
+        stability: development
+        brief: >
+          Criticality of an application, service, or workload indicates its importance to the business.
+        note: >
+          [See AppHub properties and attributes](https://cloud.google.com/app-hub/docs/overview#properties-attributes)
+        examples: ['CRITICAL', 'HIGH', 'MEDIUM', 'LOW']
+      - id: gcp.apphub.workload.id
+        type: string
+        stability: development
+        brief: >
+          The name of the workload as configured in AppHub.
+        examples: ['my-workload']
+      - id: gcp.apphub.workload.environment_type
```

Comment:
I assume [`deployment.environment.name`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/deployment.md) does not work (since there are two types - one for `service`, another one for `workload` and they may be different)?

Reply:
Yes - effectively this is the same, however as GCP is providing these labels on top of metrics, we did not want to "take away" or override user-configured environment name.

More specifically, it's possible if a user is already configuring `deployment.environment.name` then the apphub environment type may not match.

While we can/should use deployment.environment.name when possible, we need a fallback to ensure the AppHub labels don't conflict.

Additionally, as these labels are only produced by GCP itself -> did not want to rely on unstable portions of semantic conventions for them.


Again, for context, AppHub will annnotate telemetry data with these labels.  The reason for this PR is to advertise what those are and how to consume them.